src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -0,0 +1,713 @@
+
+    def get_argument_env_vars(self):
+        return self.get_param("env_vars")
+
+    def get_argument_cpu(self):
+        return self.get_param("cpu")
+
+    def get_argument_memory(self):
+        return self.get_param("memory")
+
+    def get_argument_registry_server(self):
+        return self.get_param("registry_server")
+
+    def get_argument_registry_user(self):
+        return self.get_param("registry_user")
+
+    def set_argument_registry_user(self, registry_user):
+        self.set_param("registry_user", registry_user)
+
+    def get_argument_registry_pass(self):
+        return self.get_param("registry_pass")
+
+    def set_argument_registry_pass(self, registry_pass):
+        self.set_param("registry_pass", registry_pass)
+
+    def get_argument_dapr_enabled(self):
+        return self.get_param("dapr_enabled")
+
+    def get_argument_dapr_app_port(self):
+        return self.get_param("dapr_app_port")
+
+    def get_argument_dapr_app_id(self):
+        return self.get_param("dapr_app_id")
+
+    def get_argument_dapr_app_protocol(self):
+        return self.get_param("dapr_app_protocol")
+
+    def get_argument_dapr_http_read_buffer_size(self):
+        return self.get_param("dapr_http_read_buffer_size")
+
+    def get_argument_dapr_http_max_request_size(self):
+        return self.get_param("dapr_http_max_request_size")
+
+    def get_argument_dapr_log_level(self):
+        return self.get_param("dapr_log_level")
+
+    def get_argument_dapr_enable_api_logging(self):
+        return self.get_param("dapr_enable_api_logging")
+
+    def get_argument_service_type(self):
+        return self.get_param("service_type")
+
+    def get_argument_service_bindings(self):
+        return self.get_param("service_bindings")
+
+    def get_argument_revision_suffix(self):
+        return self.get_param("revision_suffix")
+
+    def get_argument_startup_command(self):
+        return self.get_param("startup_command")
+
+    def get_argument_args(self):
+        return self.get_param("args")
+
+    def get_argument_tags(self):
+        return self.get_param("tags")
+
+    def get_argument_system_assigned(self):
+        return self.get_param("system_assigned")
+
+    def get_argument_disable_warnings(self):
+        return self.get_param("disable_warnings")
+
+    def get_argument_user_assigned(self):
+        return self.get_param("user_assigned")
+
+    def get_argument_registry_identity(self):
+        return self.get_param("registry_identity")
+
+    def get_argument_workload_profile_name(self):
+        return self.get_param("workload_profile_name")
+
+    def set_argument_workload_profile_name(self, workload_profile_name):
+        self.set_param("workload_profile_name", workload_profile_name)
+
+    def get_argument_secret_volume_mount(self):
+        return self.get_param("secret_volume_mount")
+
+    def get_argument_service_connectors_def_list(self):
+        return self.get_param("service_connectors_def_list")
```

Comment:
do we really need these methods to get arguments? Is it possible to have a general method to get argument with key and only have special get_argument_xxx() method for argument which need to be handled specifically?

Reply:
I think there are two advantages to having a method `get_argument_xxx()` rather than `self.get_param(key)`. 
1. It is convenient for users to get arguments.
2. It is not easy to make mistakes with `get_argument_xxx()`.   But with `self.get_param(key)`, if the key passed is wrong, it will lead to mistake.